lib/mongodb.js

```diff
@@ -734,8 +734,13 @@ MongoDB.prototype.buildSort = function(model, order) {
 };
 
 function buildNearFilter(query, near) {
+  var node = query.where;
+  near.key.forEach(function(key) {
+    node = node[key];
+  });
+
```

Comment:
The idea here is that `var node` holds a reference starting at the top of the `query` object with `query.where`. Each `key` (generated by the `searchForNear` function in juggler's `geo.js`) walks the reference down a "node" in the query object until it drills down to the key that has the location. Let me know if that still isn't clear or if there's a way I could code it so that it's more readable.